### ISSUE TITLE
Flag Day too far in future

### ISSUE THREAD

**Issue by nivex** [ipviolations/ipv4flagday#2]
IPv6 was first specified by RFC 2460 in December 1998. Operating system vendors have been coding to essentially this specification. This is 20 years of good work.

IPv6 was made a full Internet standard by RFC 8200 in July 2017. How much longer does something need to be a standard to be well integrated into the network? Given the rate of change of technology nowadays, I'd say five years is more than generous. I therefore suggest moving the flag day to the 2022-2024 timeframe.

Any further delay will result in excessive side effects stemming from IPv4 address exhaustion. The transfer market has already shown that gaining access to usable addresses is becoming cost prohibitive.

Related reading:
* https://www.internetsociety.org/blog/2017/07/rfc-8200-ipv6-has-been-standardized/
* https://en.wikipedia.org/wiki/IPv4_address_exhaustion

**Comment by ipviolations** [comment]
Hello @nivex, thanks for your comment on this matter. We are definitely open for discussion. We invite everyone who is interested to share comments and thoughts on this issue.

**Comment by tiagogaspar8** [comment]
I agree, It's not that difficult to deploy IPv6, actually, for example here in Portugal, 2 of the 3 ISP's support IPv6 1 of them uses IPv6 natively (without you asking for it) and the other one only enables it when asked for, so, as you can see there are ISP's that choose not to deploy IPv6, and I believe that's bad at too many levels, and ISP's are slowing down it's deployment for reasons wich I fail to understand.
Ten years is too much, as they only need to enable it with a few clicks in their management software.
I believe that if they wanted to they could deploy it in 2 years without any losses, they only need a little push.

Besides that issue, in the time that ISP's are adapting and enabling IPv6 in their infrastructures, IPv6 needs to grow a little bit with more standards, because there are things needed that IPv4 can do and IPv6 can't, mainly related with DNS. Also engineers and technicians need to have the required training, cause the IP versions are different, and their deployment methods are severely different, and several technicians believe they need NAT, Wich they don't, or tend to deploy very short prefixes to end costumers.

In conclusion there's a lot to be done, but we don't need 10 years to do it.

**Comment by DC7IA** [comment]
> How much longer does something need to be a standard to be well integrated into the network? Given the rate of change of technology nowadays, I'd say five years is more than generous.

Yup.



> I agree, It's not that difficult to deploy IPv6

Exactly, it's actually very easy, even easier than deploying legacy IP.

**Comment by DC7IA** [comment]
How about 31st December 2023?

**Comment by nivex** [comment]
The pandemic threw a wrench in a lot of things these past couple years, but I think we're starting to come out of that now.

I propose 2027-06-06 as the target date. That gives five more years from now and would be the 15th anniversary of [World IPv6 Launch](https://www.worldipv6launch.org/).

**Comment by DC7IA** [comment]
> I propose 2027-06-06 as the target date. That gives five more years from now and would be the 15th anniversary of [World IPv6 Launch](https://www.worldipv6launch.org/).

15 years should be enough, yup.

**Comment by nivex** [comment]
I'm done with this farce. Closing all my open issues.
